parquet-variant/src/variant.rs

```diff
@@ -1286,6 +1286,59 @@ impl TryFrom<(i128, u8)> for Variant<'_, '_> {
     }
 }
 
+impl std::fmt::Debug for Variant<'_, '_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Variant::Null => write!(f, "null"),
```

Comment:
I think for the debug printing we should also include the variant enum, otherwise one wouldn't be able to tell the difference between `Variant::Int8(42)` and `Variant::Int16(42)` from the debug output, as they would both be rendered as `42`

So perhaps we can keep 

```suggestion
            Variant::Null => write!(f, "Variant::Null"),
```

And then apply some nicer formatting to the binary / object ones

Reply:
I pulled out the impl Debug as its own PR that should merge first:
* https://github.com/apache/arrow-rs/pull/8140

Let's pick up the conversation there as needed, this PR will shrink a lot after rebasing on top of it.